sklearn/cross_decomposition/pls_.py

```diff
@@ -742,7 +754,10 @@ def fit(self, X, Y):
         # components is smaller than rank(X) - 1. Hence, if we want to extract
         # all the components (C.shape[1]), we have to use another one. Else,
         # let's use arpacks to compute only the interesting components.
-        if self.n_components == C.shape[1]:
+        if self.n_components > np.max(C.shape):
+            raise ValueError("n_components=%d is invalid with X of shape %s and Y of shape %s."
+                             % (self.n_components, str(X.shape), str(Y.shape)))
```

Comment:
Could you please add a new unit test or new check to an existing unit test for this case?